otter/test/convergence/test_service.py

```diff
@@ -104,24 +109,31 @@ def test_buckets_acquired(self):
         When buckets are allocated, the result of converge_all_groups is
         performed.
         """
-        def converge_all_groups(log, currently_converging, _my_buckets,
+        def converge_all_groups(currently_converging, _my_buckets,
                                 all_buckets, divergent_flags):
             return Effect(
-                ('converge-all', log, currently_converging, _my_buckets,
+                ('converge-all', currently_converging, _my_buckets,
                  all_buckets, divergent_flags))
 
         my_buckets = [0, 5]
-        sequence = SequenceDispatcher([
-            (GetChildren(CONVERGENCE_DIRTY_DIR), lambda i: ['flag1', 'flag2']),
-
+        bound_sequence = [
```

Comment:
While I do agree that having single list is nice, unfortunately it leads to deeply intended code. Especially, if its more than one level nested effects. The above would look like
```python
        sequence = SequenceDispatcher([
            (Func(uuid.uuid1), lambda i: 'uid'),
            unwrap_wrapped_effect(
                BoundFields, dict(fields={'system': 'converger',
                                          'converger_run_id': 'uid'}),
                [(GetChildren(CONVERGENCE_DIRTY_DIR),
                    lambda i: ['flag1', 'flag2']),
                (('converge-all',
                    transform_eq(lambda cc: cc is converger.currently_converging,
                                True),
                    my_buckets,
                    self.buckets,
                    ['flag1', 'flag2']),
                    lambda i: 'foo')
               ])
        ])
```
when it is sequential. It will become difficult to manage the indentation with 80 char limit. 

Sometimes, having separate `perform_` function for internal effect makes the distinction more clear. I find it easy to understand the effect code when seeing separate perform functions in test. It has direct mapping to effect code. This seems to be the case with https://github.com/rackerlabs/otter/blob/logging-effect/otter/test/convergence/test_service.py#L366 test. Sometimes they make sense inline. I am actually fine either way as long as there is not too much indentation :)